### DISCUSSION TITLE
Contributor License Agreement

### DISCUSSION THREAD

**Discussion by Olga-Yakovleva** [RHVoice/RHVoice#556] [General]
Dear contributors to the RHVoice project,

I have decided that to keep RHVoice viable we need to obtain some revenue.

One possible source of revenues is to ask for small payments from users for new “premium” features, over and above the continuing, free features we already have. In order to protect those premium features from unpaid use, I will have to publish them only as part of compiled applications for all the currently supported platforms without making the code of those features open source.

So, I need to move to a kind of “open core” model, where I have some extra, non-public code added to the code from the continuing open source repositories. Now, in order to do that, I want to obtain permission to use contributions made by others to the project.  This permission will be in the form of what is commonly called a “Contributor License Agreement’ (C.L.A.)

Essentially, the contributor retains all rights to their contribution to the project, but they license to me its use.

Very soon, any new code contributions will require a contributor to have
granted a C.L.A. to me.  For those of you who have contributed code to
RHVoice in the past, can you please contact me by email at rhvoice@rhvoice.org to discuss a C.L.A. for your past contributions? Some minor bug fixes, I am told, usually do not need a C.L.A. but it could be complicated and time-consuming to analyze every contribution. So, it will be simpler to have a C.L.A. for everything.

Important: Please note that voices are not covered by what I want to do.  I expect that if you have published a voice, you want it to continue to be provided for free on the RHVoice runtime apps.  However, if your voice has “non-commercial” in its license, be aware that this does not mean simply that it cannot be sold. Rather, it means that the voice cannot be published, even for free, by any entity which is engaged in commerce.  If RHVoice takes donations or makes sales of any kind, then it will be commercial.  As such, I will seek to obtain license from you, to publish your voice, for free, on the RHVoice apps.  

After many years, this is a change but, I feel, a step in the right direction, and necessary to cover the expenses of this growing project.

Now, to make it more likely that the contributors I'm asking to contact me
become aware of this post, I will @mention you by your GitHub user names, as is
the standard way of calling somebody on GitHub.

* @alex19EP
* @KOLANICH
* @zstanecic
* @louderpages
* @dmikushin
* @beqabeqa473
* @grzezlo
* @techtonik
* @kvark128
* @verdulo
* @TheQuinbox
* @GermanAizek 
* @alekssamos

**Comment by KOLANICH** [comment]
I'm not going to sign any CLAs because I consider CLAs as a harmful practice somehow like KYC in banks, the difference is the projects themselves choose to require CLAs.

While I understand that you'd like money, I refuse to sign any CLAs.

But you and everyone else can consider the code I have contributed to this project as distributed under [`Unlicense`](https://unlicense.org/) license where it is legally allowed.

**Reply by Olga-Yakovleva** [reply]
Thank you, @KOLANICH, for  extending the application of [Unlicense](https://unlicense.org/) to all your contributions to the RHVoice project. I understand that with [Unlicense](https://unlicense.org/) no C.L.A. is required. Thank you again!

**Comment by KOLANICH** [comment]
BTW, DCOs are also considered a harmful practice.

**Comment by KOLANICH** [comment]
BTW, if you really want to relicense RHVoice, you would have to eliminate all the dependencies under viral licenses. Like https://github.com/RHVoice/RHVoice/blob/master/src/third-party/mage/COPYING.

I also wanna mention, if RHVoice had been not under GPL, but under a permissive license, there would have been no issue to incorporate it into proprietary products without any CLAs and copyright transfers.

So I wonder if it is acceptable to relicense RHVoice to something more permissive.

**Comment by dmikushin** [comment]
@Olga-Yakovleva, I would like to proceed with the easiest possible option of your choice. If [Unlicense](https://unlicense.org/) is the easiest, hereby I consent to make all my contributions to RHVoice covered by [Unlicense](https://unlicense.org/).

Thank you all for your hard work!

**Reply by Olga-Yakovleva** [reply]
Thank you, @dmikushin! I didn't realize that there may be another option, besides a C.L.A., which contributors would be so kind to offer, until @KOLANICH replied to my post. Thank you again for applying [Unlicense](https://unlicense.org/) to your contributions to RHVoice as well!

**Comment by Olga-Yakovleva** [comment]
I have published a request for contributors with two options to give me those permissions here (permissive license like Unlicense or a traditional C.L.A.): https://github.com/RHVoice/contrib-licensing.

**Comment by zstanecic** [comment]
Hi 

@olga-yakovleva,

Link which points to the traditional CLA is not working.

**Reply by Olga-Yakovleva** [reply]
Thanks, fixed now.

**Comment by alex19EP** [comment]
I'v signed CLA

**Reply by Olga-Yakovleva** [reply]
Thank you, @alex19EP. Yes, received and dated this day.

**Comment by louderpages** [comment]
Olga,

This is to confirm that I have emailed you a signed C.L.A. today, April 23rd, 2022.

Best,
LP

**Reply by Olga-Yakovleva** [reply]
Thank you, @louderpages!

**Comment by zstanecic** [comment]
Hi @Olga-Yakovleva 
This is to confirm, that i have signed the Cla.
Although i am publishing improvements to languages and new voices, i think that it's best to have this signed, to avoid misunderstandings and bad feelings in the future.
Also, in the same e-mail please check the message, and review my Pr.

**Reply by Olga-Yakovleva** [reply]
Thank you, @zstanecic. Yes, received, dated yesterday. By the way, though I exclude contributors' own languages and voices, I am actually asking for a license to patches to my own language code. So now I can merge #554. Thank you again.

**Comment by beqabeqa473** [comment]
Hello Olga.
I hereby give notice of the signing of the cla on my part at april 24 2022

On 4/24/22, Zvonimir Stanečić ***@***.***> wrote:
> Hi @Olga-Yakovleva
> This is to confirm, that i have signed the Cla.
> Although i am publishing improvements to languages and new voices, i think
> that it's best to have this signed, to avoid misunderstandings and bad
> feelings in the future.
> Also, in the same e-mail please check the message, and review my Pr.
>
> --
> Reply to this email directly or view it on GitHub:
> https://github.com/RHVoice/RHVoice/discussions/556#discussioncomment-2622182
> You are receiving this because you were mentioned.
>
> Message ID:
> ***@***.***>


-- 
with best regards Beqa Gozalishvili
Tell: +995593454005
Email: ***@***.***
Web: https://gozaltech.org
Skype: beqabeqa473
Telegram: https://t.me/gozaltech
facebook: https://facebook.com/gozaltech
twitter: https://twitter.com/beqabeqa473
Instagram: https://instagram.com/beqa.gozalishvili

**Reply by Olga-Yakovleva** [reply]
I requested some additional steps in an email.

**Comment by Olga-Yakovleva** [comment]
Thank you, @beqabeqa473.

**Reply by alekssamos** [reply]
MIT-0

0BSD

Unlicense

**Comment by IhorShevchuk** [comment]
Hey @Olga-Yakovleva,

I have sent an email with my signed CLA today.

**Reply by Olga-Yakovleva** [reply]
Thank you, @IhorShevchuk!

**Comment by GermanAizek** [comment]
@Olga-Yakovleva, am I not late yet? I will send the CLA to your mail today.

**Reply by Olga-Yakovleva** [reply]
@GermanAizek, I have reverted #405 , but I will bring it back if you give me the license, yes, thank you!

**Comment by techtonik** [comment]
I like neither CLA, nor "open core" models as both are signs of shady corporate practices. I'd rather prefer RHVoice to be nominated for Digutal Public Good https://digitalpublicgoods.net/registry/ and have an https://opencollective.com/ to expose how underfunded these good things are, so that engineering community could see the problem with absence of sustainable feedback loops, and setup them instead of relying of the flaky busyness gameplay that was designed to keep people busy from fighting each other.

It is pity to see MAGE gone because of these shady practices - that art colab in pursue of minimal interaction latency was one awesome piece of the history I discovered while studying RHVoice code. But I realize that financial hardships is not something we can solve just by wishful thinking, at least I had to give up on that part, but I fully support your work and your decision to find a model to keep RHVoice alive. Maybe TinyML will soon replace hadwritten HMM rules, but the story behind RHVoice written by Olga Yakovleva will be an amazing tale I will hope to preserve in my memories.

All my contributions to open source are in public domain, including to RHVoice. Out of copyright "protection". So you are free to use them as you want. SPDX forces everybody to go the copyright way to feed lawyers in endless battles, so if you can use an alternative SBOM standard that respects copyright opt-outs as a "last will of an author", that will make my day better. But if not - it's fine.

**Reply by alex19EP** [reply]
> SPDX forces everybody to go the copyright way to feed lawyers in endless battles, so if you can use an alternative SBOM standard that respects copyright opt-outs as a "last will of an author", that will make my day better.

I wanted to use spdx in rhvoice... so pointing out preferred alternatives would be very helpful.

**Reply by techtonik** [reply]
CycloneDX looks more promising https://www.settletop.com/insights/understanding-sbom-standards-cyclonedx-spdx-swid but I haven't checked how did they approached the problem. SPDX statement on public domain is to treat it like a license, which it not the same as copyright opt out.

**Comment by louderpages** [comment]
Hi,

I just want to point out that MAGE is removed, but its functionality has been replaced by new code within the RHVoice core.

This new code implements the MAGE dynamic optimization functionality, and should be a little faster to execute.  So, RHVoice’s fast response time is maintained, if not improved upon.

But the main advantage is that we now have a much smaller code base to maintain,  and only one set of voice file formats to transfer and store.

**Reply by techtonik** [reply]
Code is not the point. The point is that collaboration between MAGE is what @Maria-Astrinaki and @Olga-Yakovleva is what made RHVoice possible. I don't know if they really communicated together, but both were solving problems of real people in real world that other, money hungry people could not (and probably still can't) for decades. Maria did that for systems to be interactive with people, and Olga for people to interact with systems.

https://web.archive.org/web/20170922233120/http://tcts.fpms.ac.be/~drugman/files/SLT12-Astrinaki.pdf

It is the story behind MAGE + RHVoice that is important.

**Comment by louderpages** [comment]
I really do not understand your problem.

MAGE was designed for a certain purpose.  RHVoice then made use of it.

MAGE sets the speech parameters one phoneme at a time, rather than for the whole utterance.  This permits a fast response, a quick start to speech.

Providing both the maximum quality, full optimization, and MAGE to provide the lower quality but faster option, requires two parallel sets of code within RHVoice. Take a look at the previous code base.

So. now we have modified the main, max quality HTS method to implement a per label optimization.  This now provides the same functionality as MAGE as an option to the one optimization method.

This new method is better than MAGE because:  

1. It should produce better if not similar speech quality.
2. It is faster.
3.  It cuts in half the amount of source code which needs to be modified in future.
4.  The app is smaller because it only needs the HTS format files, and not the MAGE format.

And, it is all open source.

What are you going on about? Real people’s problems, money hungry people? BTW. The copyright to MAGE is held by a for profit commercial company.

**Reply by techtonik** [reply]
I appreciate the technical description - it makes total sense. However, my point was not about technology.

"money hungry people" are voice synthesis companies. There is no market, no money, so if they want to survive, they have to do what it takes to secure money. Which often means endless development. Once you ship the product which development is sponsored by grants, public and private, your game is over.

"Real people’s problems" are problems that people using voice synthesis are experiencing daily. Slow time to response was the major one. "money hungry" voice companies (which is natural process when there is no money) were not interested to solve the high latency problem. Or they solved it, but demanded compensation. Either way, real people couldn't not get the solution, or could no afford it, so this problem was not solved for them for years (if not decades already).

I am saying that Open Source made it possible to bring multimedia MAGE project which needed low latency and RHVoice together.

Now that the link is broken, there is less art in this story. That's what I mean.

**Comment by baarkerlounger** [comment]
Is Open Core still the plan? If so would you mind sharing how you expect it to work? I.e. what would be proprietary and what would stay FOSS? Would the open part continue to be distributed on FDroid etc?

**Comment by PatrykMis** [comment]
Money? Isn't better to add donation options? NVAccess sustained this way. Personally I donate on a regular basis to projects I really use.

**Comment by louderpages** [comment]
I think that NV Access will get most of its funding from Microsoft.

**Reply by techtonik** [reply]
Are there any interviews with the guys? Open Source Sustainability problem is no joke.

**Comment by zstanecic** [comment]
NV access gets the funding from various sources, and they, for example, have money to afford the certificate for their installers.
NV access gets funding from various sponsors, including but not limited to Mozilla, Adobe, and to probably name others, which i cannot think of at the moment
Best,
Zvonimir

**Comment by PatrykMis** [comment]
It's sad. Lack of awareness in the society: people think that if something is free it is for free.

I will never ever pay for the closed-source feature of RHVoice, while I gladly donate and contribute.

There are numerous app authors who distribute free and pro (paid) version on Google Play, where pro version can be always build from the source if someone wants to - e.g. OsmAnd+.

**Reply by louderpages** [reply]
Patryk, an honorable philosophy you have there. 

I think it helpful to disentangle three different things which have been going on in this project.  I speak for myself not for Olga, I hope that I will not offend her. And I hope I will not offend you. I hope to explain more about what I see this Open Core involving.

1. There is the open source code for both building voices, and for deploying them in a run-time app. 
2. There are voices and languages being built. These can be open source in the case of languages which have actual code,  or like voices, can be copyrighted or lefted in any way you please.  People have made proprietary voices in the past. You won't find them on RHVoice apps. They are distributed elsewhere by their owners. (Not by LouderPages I must add) 
3. Then there is the building of runtime apps, making them available and supporting them. Support is the big task, as always with software because of operating system upgrades and the unfortunate fact that  every day there are users who need help. 

Any capable group has the right to perform Part 3, and incur the effort, costs and - importantly - the responsibilities involved. If they can get funding from sponsors or they spend their own money and can find the time  to do this at no cost to the users,  and without leaving users dissatisfied, then that's wonderful. 

Donations could be requested from users, but frankly the potential amounts look much less than needed.   Something more than donations is required. And, if that something can be found and the need is met,  then in that case, I personally would feel uncomfortable also asking for donations from users.

Sponsorship could solve the problem but it is hard to obtain. NV Access has been more business-like in that regard. And they have a product which benefits the likes of Microsoft and others by giving some competition to JAWS.  It is much easier to sell Microsoft Windows or Adobe Acrobat  to institutions and governments if blind users do not need to pay a huge amount for a screen-reader.

Small premium, luxury features at low prices. That is the money you are talking about and objecting to, although you may be suspecting something else, like a $50 app for purchase.  Anyway, a group could run the open source apps and charge just for maintenance, or for the fourth voice in a language, or for some other feature. Even if those features and voices are all available open source or Creative Commons.  People would pay just because they prefer to use a service they feel they can rely on and trust to be there for them the next year. If they do not want to pay $5 then the service cannot be very useful, can it? And from that we would learn that we developers should find something more valuable to do with our life.  Or, if  users cannot afford it and cannot use the app without some premium features, we can band together to get a sponsor to pay for a batch of users who need financial support.

One tricky point, though.  If you believe that voices, like books, music, spreadsheets can be copyrighted and made proprietary, then there is the problem of piracy. So, even if everything is open source, but premium voices are copyrighted, we can count on people stealing those. Not just for personal use, but for the subversive thrill of distributing the voices. This requires some sort of DRM, and that requires Open Core.

Of course, if you think something other than code should be capitalized Free in the FOSS sense, then you will see no problem, since - to update a 150 year old slogan - “Intellectual Property is Theft”. 

The rain falls from the sky. but depending on where you live you will pay for it to come out of a tap in your kitchen. Either you pay a monthly bill for its use, or your city raises taxes. Perhaps you will be lucky enough not to be a tax-payer. Or - although this is unlikely -  some engineering enthusiasts may build their own reservoirs, treatment plant and pipework and supply water to your house for free and run it in their spare time.  The tap might run dry some days, but no matter.  You can hopefully reconnect to the city supply.

**Comment by zstanecic** [comment]
yes Patryk, but how many projects live in such a way.
Osmand is dead, and no longer developed.
Please don't try to make domparations here.
This project needs support to be able to exist. All domains, server capacities and such. Yes, it is open source, but it's build on the open core model.
Read carefully what Olga said in the first post here, and all will be fine.

**Reply by techtonik** [reply]
OsmAnd repo is pretty active https://github.com/osmandapp/

Wish you guys good luck on finding the right model to back your time.

**Comment by louderpages** [comment]
Olga,

This is to confirm that I have emailed you a signed C.L.A. today, April 23rd, 2022.

Best,
LP

**Comment by PatrykMis** [comment]
@zstanecic  OsmAnd was just an example. Another one is [FairEmail](https://github.com/M66B/FairEmail)

**Comment by DarkoMilosevic86** [comment]
This is to confirm that I have emailed you a signed C.L.A. today, May 7th, 2023.

**Comment by Olga-Yakovleva** [comment]
Thank you, @DarkoMilosevic86!

**Comment by rmcpantoja** [comment]
Hi,
hereby, I confirm an e-mail with my filled CLA sent yesterday.
Thanks.

**Reply by Olga-Yakovleva** [reply]
Thank you, @rmcpantoja!

**Comment by zstanecic** [comment]
@Olga-Yakovleva The Cla bot is complaining on @rmcpantoja pulls. Maybe you forgot to add him?
